hs/src/Reach/AST/SL.hs

```diff
@@ -161,15 +162,74 @@ data SLVal
   deriving (Eq, Generic)
 
 -- | Equivalence operation on flattened or simplified structures.
-class Equiv a where
+class (Eq a) => Equiv a where
   equiv :: a -> a -> Bool
+  default equiv :: (a -> a -> Bool)
+  equiv = (==)
```

Comment:
Won't this make it so that default instances use `==` internally rather than `equiv`?

Reply:
Yes. This is probably dangerous. It allowed me to write 
```
instance Equiv Int
```

instead of 

``` 
instance Equiv Int where
  equiv = (==)
```

for a bunch of primitive stuff